conf.py

```diff
@@ -48,30 +65,54 @@
 nb_output_stderr = "remove"
 myst_heading_anchors = 3
 
-render_image_orig = NbElementRenderer.render_image
+
+# Roles “implementing” {cite}`…` and {cite:p}`…`/{cite:t}`…`
+
+
+def fake_cite(
+    name: str,
+    rawtext: str,
+    text: str,
+    lineno: int,
+    inliner: Inliner,
+    options: Mapping[str, object] = MappingProxyType({}),
+    content: Sequence[str] = (),
+) -> tuple[list[nodes.Node], list[str]]:
+    node = inliner.document.reporter.info(f"cite:{text}")
+    return [node], []
+
+
+class FakeDomain(Domain):
+    name = "cite"
+    roles = dict(p=fake_cite, t=fake_cite)
+
+
+# Role linking to the canonical location in scanpy’s docs
 
 
-def render_image(self: NbElementRenderer, data: MimeData) -> list[nodes.Element]:
-    """Makes images display size default to their jupyter setting.
+class CanonicalTutorial(SphinxDirective):
+    required_arguments = 1
+    has_content = False
 
-    Workaround for: https://github.com/executablebooks/MyST-NB/issues/522
-    """
-    node_list = render_image_orig(self, data)
-    try:
-        [image] = node_list
-        if not isinstance(image, nodes.image):
-            raise ValueError(f"Expected nodes.image, got {type(image)}")
-        for key in ("width", "height"):
-            if key in image:
-                continue
-            if (v := data.output_metadata.get(data.mime_type, {}).get(key)) is None:
-                continue
-            image[key] = str(v)
-    except Exception:
-        return node_list
-    return [image]
+    def run(self) -> list[nodes.Node]:
+        text = self.arguments[0]
+        ref = resolve_reference_in_inventory(
+            self.env,
+            "scanpy",
+            addnodes.pending_xref("", reftype="doc", refdomain="std", reftarget=text),
+            nodes.inline("", text),
+        )
```

Comment:
I can't find this on the `intersphinx` docs page so I really can't say what it does: https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#module-sphinx.ext.intersphinx also googling gave me nothing

Reply:
This makes this

```
:::{canonical-tutorial} tutorials/experimental/dask
:::
```

render into this:

<img width="809" alt="image" src="https://github.com/scverse/scanpy-tutorials/assets/291575/964e1b45-424e-4316-914a-cf5594c70035">

Reply:
the function can return `None` which I catch with the `assert`. So the docs don’t build if one of these tries to link with a doc that isn’t in scanpy.

Reply:
I think my point is that a comment might be nice so that the next person can know what with function is doing.

Reply:
The docstring helps, but sure.

![grafik](https://github.com/scverse/scanpy-tutorials/assets/291575/17ede3fd-5009-4f04-9fd1-89e35149c646)